sys/riscv/riscv/freebsd64_machdep.c

```diff
@@ -246,7 +246,7 @@ freebsd64_sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	onstack = sigonstack(tf->tf_sp);
 
 	CTR4(KTR_SIG, "sendsig: td=%p (%s) catcher=%p sig=%d", td, p->p_comm,
-	    catcher, sig);
+	    (vaddr_t)catcher, sig);
```

Comment:
Does this need to be `(__cheri_addr vaddr_t)`?  I think `sig_t` is a capability in both hybrid and purecap?

Reply:
I thought `__cheri_addr` was only necessary to avoid the offset interpretation?  I could be confused.

Reply:
I think it isn't yet optional?  At least, I get warnings if I try to not use one of `__cheri_addr` or `__cheri_offset` when casting a capability to a scalar.

Reply:
Yeah unfortunately we still get a ctoptr. I've started to add a new flag to always return the address for those casts but I'll only be able to finish it after one finished the current merge.